pkg/controller/deployment/deployment_controller.go

```diff
@@ -562,6 +563,14 @@ func (dc *DeploymentController) getNewRC(deployment extensions.Deployment) (*api
 			Template: &newRCTemplate,
 		},
 	}
+	if deployment.Annotations != nil {
```

Comment:
You can still access a nil map w/o panicing (assigning to a nil map is the only way to make it panic). Remove this if to reduce indentation.